### PR TITLE
incubator/vault Add vault api address for non dev setup

### PR DESCRIPTION
**Pull request by devdinu**
Signed-off by: Dinesh Kumar <dineshkumar_cse@hotmail.com>

#### What this PR does / why we need it:
* Enables vault to be installed in non dev mode
chart: incubator/vault

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by devdinu** [comment]
I would also love to add sample `overrides.yaml` / doc for vault cluster setup, you can suggest the right place.

File: incubator/vault/templates/deployment.yaml [hunk 1]
```diff
@@ -95,8 +95,13 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          - name: VAULT_API_ADDR
+            value: "https://$(POD_IP):8200"
+        {{- if .Values.vault.dev }}
```
Comment (jbialy) [inline incubator/vault/templates/deployment.yaml]: Instead of using an empty if condition, how about using `if not .Values.vault.dev` instead?

Comment (devdinu) [reply incubator/vault/templates/deployment.yaml]: I've updated the PR.

**Pull request merged**